File: src/main/java/susc/s01/Data/Log/LogData.java
package susc.s01.Data.Log;

import susc.s01.Data.Log.JoinLog.PlayerLog;

import java.util.UUID;
import java.util.concurrent.ConcurrentHashMap;

public class LogData {
    private static class PlayerLogData {
        private final static ConcurrentHashMap<UUID, PlayerLog> INSTANCE = new ConcurrentHashMap<>();
    }

    public static ConcurrentHashMap<UUID, PlayerLog> getJoinLogDataInstance() {
        return PlayerLogData.INSTANCE;
    }
}

File: src/main/java/susc/s01/Data/Log/LogData.kt
package susc.s01.Data.Log

import susc.s01.Data.Log.JoinLog.PlayerLog
import java.util.*
import java.util.concurrent.ConcurrentHashMap

object LogData {
    private object PlayerLogData {
        val joinLogDataInstance: ConcurrentHashMap<UUID, PlayerLog> = ConcurrentHashMap()
            get() = PlayerLogData.INSTANCE
    }
}
